### PR TITLE
Bracket docs

### LINKED ISSUE #1129: [Arrow] Bracket docs
`Bracket` was added in #34 and merged to unlock work over other related Type classes, but docs are still not available. We should work on them asap.

### PR DESCRIPTION
**Pull request by JorgeCastilloPrz**
fixes #1129

### CODE REVIEW

File: modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+## Bracket
+
+{:.intermediate}
+intermediate
+
+The `Bracket` Type class abstracts the ability to safely **acquire**, **use**, and then **release** a resource. 
+
+Essentially, it could be considered the functional programming equivalent to the well known imperative
```
[the remark this answers is not in the document]
Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md]: Fixed. I've just added an additional example clarifying that and showcasing a polymorphic program using `Bracket<F, Throwable>` as a constraint, and running over the 3 mentioned data types.

File: modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+Essentially, it could be considered the functional programming equivalent to the well known imperative 
+`try/catch/finally` structure.
+
+`Bracket` extends [`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }}).
+
+It ensures that the acquired resource is released at the end after using it, **even if the using action throws an error**. 
+That ensures that no errors are swallowed.
+
+Another key point of `Bracket` would be the ability to abstract over whether the resource is going to be used
```
[the remark this answers is not in the document]
Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md]: Fixed, explained above.

File: modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ 
+Read the [`IO`]({{ '/docs/effects/io' | relative_url }}) docs for more context on this.
+ 
+Now, let's say we want to open a file, do some work with it, and then close it. With `Bracket`, we could make that 
+process look like this:
+
+```kotlin:ank
+import arrow.effects.IO
+
+openFile("data.json").bracket(release = { closeFile(it) }) { file ->
```
[the remark this answers is not in the document]
Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md]: Fixed now.

File: modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+It's a generalized version of `bracket()` which uses `ExitCase` to distinguish between different exit cases when 
+releasing the acquired resource. `ExitCase` can take the values `Completed`, `Canceled`, or `Error(e)`.  So depending 
+how the `use` execution finalizes, the corresponding `ExitCase` value will be passed to the `release` lambda.
+
+Requires passing `release` and `use` lambdas. It ensures acquiring, using and releasing the resource at the end.
+
+`fun <A, B> Kind<F, A>.bracketCase(release: (A, ExitCase<Throwable>) -> Kind<F, Unit>, use: (A) -> Kind<F, B>): Kind<F, B>`
+
+```kotlin:ank
```
[the remark this answers is not in the document]
Comment (JorgeCastilloPrz) [reply modules/docs/arrow-docs/docs/docs/typeclasses/bracket/README.md]: Sure, I'll take a look tomorrow. Leaving now!

**Comment by JorgeCastilloPrz** [comment]
@raulraja tried to do port everything to be executable but `Bracket` was included in `0.8.1` and try arrow is still not pointing to that version. Would it be okay to merge the docs as they are and iterate on that once try arrow gets upgraded?

**Comment by calvellido** [comment]
I think the problem is not related with the Arrow version, but with some changes needed to be made on the compiler config as the error some of those snippets are facing is the following:

`
Exception in thread "main" java.lang.IllegalStateException: No BuiltInsLoader implementation was found. Please ensure that the META-INF/services/ is not stripped from your application and that the Java virtual machine is not running under a security manager
`

**Pull request merged**